appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmRuntimeFileLogHandler.java

```diff
@@ -102,4 +111,57 @@ private static void reloadLoggingProperties(LogManager logManager) {
       System.err.println(e.getClass().getName() + ": " + e.getMessage());
     }
   }
+
+  @Override
+  public void close() throws SecurityException {
+    super.close();
+    if (handlers.decrementAndGet() <= 0 ) {
+      queue.offer(null);
+    }
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    if (isLoggable(record)) {
+      queue.offer(new LogEntry(record));
+    }
+  }
+
+  private void superPublish(LogRecord record) {
+    super.publish(record);
+  }
+
+  private class LogEntry {
```

Comment:
What's the reason for introducing this class? Can the queue simply hold LogRecords directly?